src/MainWindow.cpp

```diff
@@ -73,7 +77,8 @@ MainWindow::MainWindow(BaseObjectType* cobject, Glib::RefPtr<Gtk::Builder> const
 
     show_all();
 
-    m_trayIcon.setVisible(Settings::instance().closeToTray());
+    m_trayIcon.setVisible(Settings::instance().closeToTray() || Settings::instance().startAtTray());
```

Comment:
No, tray icon's status should not depend on the status of start in tray. Instead, start in tray should read its value from settings and can only be sensitive if closed to tray is active. So; you should set its sensitivity and status to false in onCloseToTray if closed to tray becomes inactive and save the settings in `onStartInTray` as you did. In that case, onStartInTray will be fired naturally and the setting will be saved.

Reply:
`m_trayIcon.setVisible` sets visibility of tray icon. We should make tray icon visible if either closeToTray or startInTray is set to True. I see nothing wrong here.

```cpp
    m_trayIcon.setVisible(Settings::instance().closeToTray() || Settings::instance().startInTray());
    startInTraySwitch->set_state(Settings::instance().startInTray());
    closeToTraySwitch->set_state(Settings::instance().closeToTray());
```
Can you eleborate more what is wrong with these lines?